types/src/transaction/signature_verified_transaction.rs

```diff
@@ -124,6 +124,29 @@ impl BlockExecutableTransaction for SignatureVerifiedTransaction {
     ) -> Self {
         Transaction::block_epilogue_v1(block_id, block_end_info, fee_distribution).into()
     }
+
+    fn pre_write_values(&self) -> Vec<(Self::Key, Self::Value)> {
+        if let Some(timestamp) = match self {
```

Comment:
nit/q: could this be extracted out, or matches!/early return or smt to avoid non-exhaustive match (although here it's fine to do optimization for whatever subset so non-exhaustive seems fine)